bin/review-rot

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python
+
+import os
+import logging
+import argparse
+import yaml
+from os.path import expanduser
+from reviewrot import get_git_service
+
+
+def main(state_, value, duration):
+    """
+    Reads input configuration file.
+    Calls appropriate git service with suitable inputs.
+
+    state_ (str): The filter(state) for review requests, e.g, older or newer
+    value (int): The value in terms of duration for requests
+                 to be older or newer than
+    duration (str): The duration in terms of period(year,month,hour,minute)
+                    for requests to be older or newer than.
+    """
+    log = logging.getLogger('review-rot')
+    home = expanduser('~')
+
+    path_to_config = os.path.join(home, '.reviewrot.yaml')
+    if not os.path.exists(path_to_config):
+        raise RuntimeError("No config '.reviewrot.yaml' provided. Please"
+                           " specify it in home directory: %s" % home)
+    # read input from home directory for pull requests
+    with open(path_to_config, 'r') as f:
+        config = yaml.load(f)
+
+    for item in config:
+        if not item.has_key('type'):
+            log.debug('git service type not found for %s', item)
+            raise KeyError('git service not found for %s' % item)
+
+        # get git service
+        git_service = get_git_service(item['type'])
+        """
+        check if username and/or repository information is given for
+        specified git service
+        """
+        if item['repos'] is not None:
+            # for each input call specified git service
+            for data in item['repos']:
+                """
+                split and format username and repository name to further
+                request pull requests
+                """
+                res = format_user_repo_name(data)
+                """
+                get pull requests for specified git service, username
+                and repository name
+                """
+                if not item.has_key('token'):
+                    log.debug('token not found for %s', item['type'])
+                    raise KeyError("token not found for %s. Please provide 'None' "
```

Comment:
How about doing this instead:

```python
token = item.get('token')
host = item.get('host')
```

If the user leaves the fields out it defaults the value to `None`. Likewise, if they set the field to `null` it has the same effect. No need to fail as this can be implied behavior.